tasks/run-file-updates/run-file-updates.yaml

```
@@ -76,26 +88,30 @@ spec:
                            -p ref="${ref}" \
                            -p paths="${updatedPaths}" \
                            -p application=${application} \
+                           -p file_updates_secret="${file_updates_secret}" \
                            -s "$(params.synchronously)" \
-                           -l ${pipelinerun_label}=$(params.pipelineRunUid) \
-                           > $(workspaces.data.path)/ir-result.txt || \
-                           (grep "^\[" $(workspaces.data.path)/ir-result.txt | jq . && exit 1)
-          
-          internalRequest=$(awk 'NR==1{ print $2 }' $(workspaces.data.path)/ir-result.txt | xargs)
-          echo "done (${internalRequest})"
+                           -l ${TASK_LABEL}="${TASK_ID}" \
+                           -l "${PIPELINERUN_LABEL}=$(params.pipelineRunUid)"
 
-          results=$(kubectl get internalrequest $internalRequest -o=jsonpath='{.status.results}')
-          if [ $(jq -jr '.buildState' <<< "${results}") == "Failed" ]; then
+          IR=$(kubectl get internalrequest \
+              -l "${PIPELINERUN_LABEL}=$(params.pipelineRunUid),${TASK_LABEL}=${TASK_ID}" \
+              --no-headers -o custom-columns=":metadata.name" --sort-by=.metadata.creationTimestamp | head -1)
+          IRjson=$(kubectl get "internalrequest/${IR}" -ojson)
```

Nit: If you switch these, you save one `kubectl` call.
```suggestion
          IRjson=$(kubectl get internalrequest \
              -l "${PIPELINERUN_LABEL}=$(params.pipelineRunUid),${TASK_LABEL}=${TASK_ID}" \
              -o jsonpath='{.items[0]}' --sort-by=.metadata.creationTimestamp )
          IR=$(jq -r '.metadata.name' "${IRjson}")
```
(Or maybe you don't even need the name?)

---

updated. thanks!